src/main/java/de/unijena/cheminf/mortar/gui/controls/ToggleSwitch.java

```diff
@@ -0,0 +1,196 @@
+import javafx.animation.TranslateTransition;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.scene.control.Control;
+import javafx.scene.effect.DropShadow;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Circle;
+import javafx.scene.shape.Rectangle;
+import javafx.util.Duration;
+
+/**
+ * A toggle switch to en- and disable features in settings.
+ *
+ * @author Zeynep Dagtekin
+ * @version 1.0.0.0
+ */
+public class ToggleSwitch extends Control {
+    //<editor-fold desc="private final class constants" defaultstate="collapsed">
+    /**
+     * Button.
+     */
+    private final Circle switchButton;
+    /**
+     * Background of the switch.
+     */
+    private final Rectangle switchBackground;
+    /**
+     * Boolean property to keep track of the state.
+     */
+    private final SimpleBooleanProperty switchedOn;
+    /**
+     * transition of the Circle from one side to the other.
+     */
+    private final TranslateTransition switchAnimation;
+    /**
+     * Color transition.
+     */
+    private final FillTransition fillAnimation;
+    /**
+     * Parallel transition of color and Circle.
+     */
+    private final ParallelTransition switchTransition;
+    //</editor-fold>
+    /**
+     * Constructor.
+     */
+    public ToggleSwitch() {
+        /**
+         * code inspired by "JavaFX UI: iOS Style Toggle Switch", uploaded by Almas Baimagambetov on YouTube
+         * https://youtu.be/maX5ymmQixM?si=v2ULa57-pjCmoQlf, 05/17/2024, 10:33
+         */
+        super();
+        this.switchedOn = new SimpleBooleanProperty(false);
+        this.switchBackground = new Rectangle(45, 18);
+        this.switchBackground.setArcWidth(18);
+        this.switchBackground.setArcHeight(18);
+        this.switchBackground.setLayoutX(-50);
+        this.switchBackground.setFill(Color.LIGHTGRAY);
+        this.switchBackground.setStroke(Color.DARKGRAY);
+        this.switchButton = new Circle(10);
+        this.switchButton.setCenterX(-40);
+        this.switchButton.setCenterY(9);
+        this.switchButton.setFill(Color.WHITE);
+        this.switchButton.setStroke(Color.DARKGRAY);
+        this.switchButton.setEffect(new DropShadow(5, Color.GRAY));
+        this.switchAnimation = new TranslateTransition(Duration.seconds(0.25));
+        this.switchAnimation.setNode(this.switchButton);
+        this.fillAnimation = new FillTransition(Duration.seconds(0.25));
+        this.switchTransition = new ParallelTransition(this.switchAnimation, this.fillAnimation);
+        this.switchAnimation.setNode(this.switchButton);
+        this.fillAnimation.setShape(this.switchBackground);
+        getChildren().addAll(this.switchBackground, this.switchButton);
+        //Listener
+        this.switchedOn.addListener((observable, oldValue, newValue) -> {
+            boolean tmpIsOn = newValue.booleanValue();
+            this.switchAnimation.setToX(tmpIsOn ? (44 - 18) : 0);
```

Comment:
These values refer to the rectangle sizes, right? Then please query them from the rectangle instead of writing them explicitly again.